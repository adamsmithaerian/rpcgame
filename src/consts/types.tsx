export enum Shape {
  Rock,
  Paper,
  Scissors
}

export enum Result {
  Win,
  Lose,
  Tie
}

export const shapeNames: Array<String> = ['Rock', 'Paper', 'Scissors'];

export const resultNames: Array<String> = ['Win', 'Lose', 'Tie'];
