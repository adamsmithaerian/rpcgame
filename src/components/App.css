span {
  margin: 10px;
}

button {
  margin: 10px;
}

input[type=submit] {
  margin: 10px;
}

select {
  margin: 10px;
}

* {
  box-sizing: border-box;
}

.text {
  padding: 10px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}