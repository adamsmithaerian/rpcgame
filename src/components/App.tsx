import * as React from 'react';
import ShapeSelectForm from './ShapeSelectForm';
import ResultPannel from './ResultPannel';
import { Shape, Result } from '../consts/types';

import './App.css';

interface AppProps {
  computer: {
    selectedShape: Shape;
  };
  user: {
    selectedShape: Shape;
  };
  onSelectShape: () => void;
  onSubmitShape: () => void;
  onResetGame: () => void;
  showResult: boolean;
  result: Result;
}

class App extends React.Component<AppProps, any> {
  renderContent() {
    if (!this.props.showResult) {
      return (
        <ShapeSelectForm
          onSelectShape={this.props.onSelectShape}
          onSubmitShape={this.props.onSubmitShape}
          selectedShape={this.props.user.selectedShape}
        />        
      );
    } else {
      return (
        <ResultPannel
          result={this.props.result}
          userShape={this.props.user.selectedShape}
          computerShape={this.props.computer.selectedShape}
          onResetGame={this.props.onResetGame}
        />
      );
    }
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">
            Rock Paper Scissors
          </h1>
        </header>
        <div className="App-intro">
          {this.renderContent()}
        </div>
      </div>
    );
  }
}

export default App;
