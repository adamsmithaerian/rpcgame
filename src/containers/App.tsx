import { connect } from 'react-redux';
import { setSelectShape, submitShape, resetGame } from '../actions';
import { Shape } from '../consts/types';

import App from '../components/App';

const mapStateToProps = (state: any) => ({
  computer: state.computer,
  user: state.user,
  result: state.app.result,
  showResult: state.app.showResult
});

const mapDispatchToProps = (dispatch: any) => ({
  onSelectShape: (shape: Shape) => dispatch(setSelectShape(shape)),
  onSubmitShape: () => dispatch(submitShape()),
  onResetGame: () => dispatch(resetGame())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
