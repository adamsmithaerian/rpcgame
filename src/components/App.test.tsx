import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './App';
import { Shape, Result } from '../consts/types';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <App
      computer={{
        selectedShape: Shape.Rock
      }}
      user={{
        selectedShape: Shape.Rock
      }}
      onSelectShape={() => {}}
      onSubmitShape={() => {}}
      onResetGame={() => {}}
      showResult={false}
      result={Result.Win}
    />,
    div
  );
});
