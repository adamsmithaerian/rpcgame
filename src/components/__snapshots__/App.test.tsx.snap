// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Rock Paper Scissors
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="text"
      >
        Please select your shape
      </label>
      <div
        className="Shape-select"
      >
        <select
          onChange={[Function]}
          value={0}
        >
          <option
            value={0}
          >
            Rock
          </option>
          <option
            value={1}
          >
            Paper
          </option>
          <option
            value={2}
          >
            Scissors
          </option>
        </select>
      </div>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`App component should matche snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Rock Paper Scissors
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="text"
      >
        Please select your shape
      </label>
      <div
        className="Shape-select"
      >
        <select
          onChange={[Function]}
          value={0}
        >
          <option
            value={0}
          >
            Rock
          </option>
          <option
            value={1}
          >
            Paper
          </option>
          <option
            value={2}
          >
            Scissors
          </option>
        </select>
      </div>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`App component should matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Rock Paper Scissors
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="text"
      >
        Please select your shape
      </label>
      <div
        className="Shape-select"
      >
        <select
          onChange={[Function]}
          value={0}
        >
          <option
            value={0}
          >
            Rock
          </option>
          <option
            value={1}
          >
            Paper
          </option>
          <option
            value={2}
          >
            Scissors
          </option>
        </select>
      </div>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Rock Paper Scissors
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="text"
      >
        Please select your shape
      </label>
      <div
        className="Shape-select"
      >
        <select
          onChange={[Function]}
          value={0}
        >
          <option
            value={0}
          >
            Rock
          </option>
          <option
            value={1}
          >
            Paper
          </option>
          <option
            value={2}
          >
            Scissors
          </option>
        </select>
      </div>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
